File: client/iih_stub/iih.ci_stub/iih.ci.ord/cior/d/OrdApOutDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;
using System.Collections.Generic;

namespace iih.ci.ord.cior.d
{
    /// <summary>
    /// OrdApOutDO 的摘要说明。
    /// </summary>
    public class OrdApOutDO : BaseDO {

        public OrdApOutDO() {
            this.Fg_death = false;
            this.Fg_autopsy = false;
        }
		public string Id_orout {
            get { return getAttrVal<string>("Id_orout",null); }
            set { setAttrVal<string>("Id_orout", value); }
        }
		public string Id_or {
            get { return getAttrVal<string>("Id_or",null); }
            set { setAttrVal<string>("Id_or", value); }
        }
		public string Id_outtp {
            get { return getAttrVal<string>("Id_outtp",null); }
            set { setAttrVal<string>("Id_outtp", value); }
        }
		public string Sd_outtp {
            get { return getAttrVal<string>("Sd_outtp",null); }
            set { setAttrVal<string>("Sd_outtp", value); }
        }
		public DateTime? Dt_timeout {
            get { return getAttrVal<FDateTime>("Dt_timeout",null); }
            set { setAttrVal<FDateTime>("Dt_timeout", value); }
        }
		public string Id_dep_out {
            get { return getAttrVal<string>("Id_dep_out",null); }
            set { setAttrVal<string>("Id_dep_out", value); }
        }
		public string Id_dep_nur_out {
            get { return getAttrVal<string>("Id_dep_nur_out",null); }
            set { setAttrVal<string>("Id_dep_nur_out", value); }
        }
		public string Id_bed_out {
            get { return getAttrVal<string>("Id_bed_out",null); }
            set { setAttrVal<string>("Id_bed_out", value); }
        }
		public bool? Fg_death {
            get { return getAttrVal<FBoolean>("Fg_death",false); }
            set { setAttrVal<FBoolean>("Fg_death", value); }
        }
		public bool? Fg_autopsy {
            get { return getAttrVal<FBoolean>("Fg_autopsy",false); }
            set { setAttrVal<FBoolean>("Fg_autopsy", value); }
        }
		public string Des_outtp {
            get { return getAttrVal<string>("Des_outtp",null); }
            set { setAttrVal<string>("Des_outtp", value); }
        }
		public bool? Fg_again31 {
            get { return getAttrVal<FBoolean>("Fg_again31",null); }
            set { setAttrVal<FBoolean>("Fg_again31", value); }
        }
		public string Des_again31 {
            get { return getAttrVal<string>("Des_again31",null); }
            set { setAttrVal<string>("Des_again31", value); }
        }
		public string Def1 {
            get { return getAttrVal<string>("Def1",null); }
            set { setAttrVal<string>("Def1", value); }
        }
		public string Def2 {
            get { return getAttrVal<string>("Def2",null); }
            set { setAttrVal<string>("Def2", value); }
        }
		public string Def3 {
            get { return getAttrVal<string>("Def3",null); }
            set { setAttrVal<string>("Def3", value); }
        }
		public string Def4 {
            get { return getAttrVal<string>("Def4",null); }
            set { setAttrVal<string>("Def4", value); }
        }
		public string Def5 {
            get { return getAttrVal<string>("Def5",null); }
            set { setAttrVal<string>("Def5", value); }
        }
		public string Def6 {
            get { return getAttrVal<string>("Def6",null); }
            set { setAttrVal<string>("Def6", value); }
        }
		public string Def7 {
            get { return getAttrVal<string>("Def7",null); }
            set { setAttrVal<string>("Def7", value); }
        }
		public string Def8 {
            get { return getAttrVal<string>("Def8",null); }
            set { setAttrVal<string>("Def8", value); }
        }
		public string Def9 {
            get { return getAttrVal<string>("Def9",null); }
            set { setAttrVal<string>("Def9", value); }
        }
		public string Def10 {
            get { return getAttrVal<string>("Def10",null); }
            set { setAttrVal<string>("Def10", value); }
        }
		public string Createdby {
            get { return getAttrVal<string>("Createdby",null); }
            set { setAttrVal<string>("Createdby", value); }
        }
		public DateTime? Createdtime {
            get { return getAttrVal<FDateTime>("Createdtime",null); }
            set { setAttrVal<FDateTime>("Createdtime", value); }
        }
		public string Modifiedby {
            get { return getAttrVal<string>("Modifiedby",null); }
            set { setAttrVal<string>("Modifiedby", value); }
        }
		public DateTime? Modifiedtime {
            get { return getAttrVal<FDateTime>("Modifiedtime",null); }
            set { setAttrVal<FDateTime>("Modifiedtime", value); }
        }
		public string Name_outtp {
            get { return getAttrVal<string>("Name_outtp",null); }
            set { setAttrVal<string>("Name_outtp", value); }
        }
		public string Name_dep_out {
            get { return getAttrVal<string>("Name_dep_out",null); }
            set { setAttrVal<string>("Name_dep_out", value); }
        }
		public string Name_dep_nur_out {
            get { return getAttrVal<string>("Name_dep_nur_out",null); }
            set { setAttrVal<string>("Name_dep_nur_out", value); }
        }
		public string Name_bed_out {
            get { return getAttrVal<string>("Name_bed_out",null); }
            set { setAttrVal<string>("Name_bed_out", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_ap_out";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_orout";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.ord.cior.d.OrdApOutDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_orout", "Id_or", "Id_outtp", "Sd_outtp", "Dt_timeout", "Id_dep_out", "Id_dep_nur_out", "Id_bed_out", "Fg_death", "Fg_autopsy", "Des_outtp", "Fg_again31", "Des_again31", "Def1", "Def2", "Def3", "Def4", "Def5", "Def6", "Def7", "Def8", "Def9", "Def10", "Createdby", "Createdtime", "Modifiedby", "Modifiedtime", "Name_outtp", "Name_dep_out", "Name_dep_nur_out", "Name_bed_out"};
        }
        
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/cior/d/OrdApOutDO.java
package iih.ci.ord.cior.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.ord.cior.d.desc.OrdApOutDODesc;
import java.math.BigDecimal;

/**
 * 医嘱出院申请 DO数据 
 * 
 */
public class OrdApOutDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_OROUT= "Id_orout";
	public static final String ID_OR= "Id_or";
	public static final String ID_OUTTP= "Id_outtp";
	public static final String SD_OUTTP= "Sd_outtp";
	public static final String DT_TIMEOUT= "Dt_timeout";
	public static final String ID_DEP_OUT= "Id_dep_out";
	public static final String ID_DEP_NUR_OUT= "Id_dep_nur_out";
	public static final String ID_BED_OUT= "Id_bed_out";
	public static final String FG_DEATH= "Fg_death";
	public static final String FG_AUTOPSY= "Fg_autopsy";
	public static final String DES_OUTTP= "Des_outtp";
	public static final String FG_AGAIN31= "Fg_again31";
	public static final String DES_AGAIN31= "Des_again31";
	public static final String DEF1= "Def1";
	public static final String DEF2= "Def2";
	public static final String DEF3= "Def3";
	public static final String DEF4= "Def4";
	public static final String DEF5= "Def5";
	public static final String DEF6= "Def6";
	public static final String DEF7= "Def7";
	public static final String DEF8= "Def8";
	public static final String DEF9= "Def9";
	public static final String DEF10= "Def10";
	public static final String CREATEDBY= "Createdby";
	public static final String CREATEDTIME= "Createdtime";
	public static final String MODIFIEDBY= "Modifiedby";
	public static final String MODIFIEDTIME= "Modifiedtime";
	public static final String NAME_OUTTP= "Name_outtp";
	public static final String NAME_DEP_OUT= "Name_dep_out";
	public static final String NAME_DEP_NUR_OUT= "Name_dep_nur_out";
	public static final String NAME_BED_OUT= "Name_bed_out";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_orout() {
		return ((String) getAttrVal("Id_orout"));
	}	
	public void setId_orout(String Id_orout) {
		setAttrVal("Id_orout", Id_orout);
	}
	public String getId_or() {
		return ((String) getAttrVal("Id_or"));
	}	
	public void setId_or(String Id_or) {
		setAttrVal("Id_or", Id_or);
	}
	public String getId_outtp() {
		return ((String) getAttrVal("Id_outtp"));
	}	
	public void setId_outtp(String Id_outtp) {
		setAttrVal("Id_outtp", Id_outtp);
	}
	public String getSd_outtp() {
		return ((String) getAttrVal("Sd_outtp"));
	}	
	public void setSd_outtp(String Sd_outtp) {
		setAttrVal("Sd_outtp", Sd_outtp);
	}
	public FDateTime getDt_timeout() {
		return ((FDateTime) getAttrVal("Dt_timeout"));
	}	
	public void setDt_timeout(FDateTime Dt_timeout) {
		setAttrVal("Dt_timeout", Dt_timeout);
	}
	public String getId_dep_out() {
		return ((String) getAttrVal("Id_dep_out"));
	}	
	public void setId_dep_out(String Id_dep_out) {
		setAttrVal("Id_dep_out", Id_dep_out);
	}
	public String getId_dep_nur_out() {
		return ((String) getAttrVal("Id_dep_nur_out"));
	}	
	public void setId_dep_nur_out(String Id_dep_nur_out) {
		setAttrVal("Id_dep_nur_out", Id_dep_nur_out);
	}
	public String getId_bed_out() {
		return ((String) getAttrVal("Id_bed_out"));
	}	
	public void setId_bed_out(String Id_bed_out) {
		setAttrVal("Id_bed_out", Id_bed_out);
	}
	public FBoolean getFg_death() {
		return ((FBoolean) getAttrVal("Fg_death"));
	}	
	public void setFg_death(FBoolean Fg_death) {
		setAttrVal("Fg_death", Fg_death);
	}
	public FBoolean getFg_autopsy() {
		return ((FBoolean) getAttrVal("Fg_autopsy"));
	}	
	public void setFg_autopsy(FBoolean Fg_autopsy) {
		setAttrVal("Fg_autopsy", Fg_autopsy);
	}
	public String getDes_outtp() {
		return ((String) getAttrVal("Des_outtp"));
	}	
	public void setDes_outtp(String Des_outtp) {
		setAttrVal("Des_outtp", Des_outtp);
	}
	public FBoolean getFg_again31() {
		return ((FBoolean) getAttrVal("Fg_again31"));
	}	
	public void setFg_again31(FBoolean Fg_again31) {
		setAttrVal("Fg_again31", Fg_again31);
	}
	public String getDes_again31() {
		return ((String) getAttrVal("Des_again31"));
	}	
	public void setDes_again31(String Des_again31) {
		setAttrVal("Des_again31", Des_again31);
	}
	public String getDef1() {
		return ((String) getAttrVal("Def1"));
	}	
	public void setDef1(String Def1) {
		setAttrVal("Def1", Def1);
	}
	public String getDef2() {
		return ((String) getAttrVal("Def2"));
	}	
	public void setDef2(String Def2) {
		setAttrVal("Def2", Def2);
	}
	public String getDef3() {
		return ((String) getAttrVal("Def3"));
	}	
	public void setDef3(String Def3) {
		setAttrVal("Def3", Def3);
	}
	public String getDef4() {
		return ((String) getAttrVal("Def4"));
	}	
	public void setDef4(String Def4) {
		setAttrVal("Def4", Def4);
	}
	public String getDef5() {
		return ((String) getAttrVal("Def5"));
	}	
	public void setDef5(String Def5) {
		setAttrVal("Def5", Def5);
	}
	public String getDef6() {
		return ((String) getAttrVal("Def6"));
	}	
	public void setDef6(String Def6) {
		setAttrVal("Def6", Def6);
	}
	public String getDef7() {
		return ((String) getAttrVal("Def7"));
	}	
	public void setDef7(String Def7) {
		setAttrVal("Def7", Def7);
	}
	public String getDef8() {
		return ((String) getAttrVal("Def8"));
	}	
	public void setDef8(String Def8) {
		setAttrVal("Def8", Def8);
	}
	public String getDef9() {
		return ((String) getAttrVal("Def9"));
	}	
	public void setDef9(String Def9) {
		setAttrVal("Def9", Def9);
	}
	public String getDef10() {
		return ((String) getAttrVal("Def10"));
	}	
	public void setDef10(String Def10) {
		setAttrVal("Def10", Def10);
	}
	public String getCreatedby() {
		return ((String) getAttrVal("Createdby"));
	}	
	public void setCreatedby(String Createdby) {
		setAttrVal("Createdby", Createdby);
	}
	public FDateTime getCreatedtime() {
		return ((FDateTime) getAttrVal("Createdtime"));
	}	
	public void setCreatedtime(FDateTime Createdtime) {
		setAttrVal("Createdtime", Createdtime);
	}
	public String getModifiedby() {
		return ((String) getAttrVal("Modifiedby"));
	}	
	public void setModifiedby(String Modifiedby) {
		setAttrVal("Modifiedby", Modifiedby);
	}
	public FDateTime getModifiedtime() {
		return ((FDateTime) getAttrVal("Modifiedtime"));
	}	
	public void setModifiedtime(FDateTime Modifiedtime) {
		setAttrVal("Modifiedtime", Modifiedtime);
	}
	public String getName_outtp() {
		return ((String) getAttrVal("Name_outtp"));
	}	
	public void setName_outtp(String Name_outtp) {
		setAttrVal("Name_outtp", Name_outtp);
	}
	public String getName_dep_out() {
		return ((String) getAttrVal("Name_dep_out"));
	}	
	public void setName_dep_out(String Name_dep_out) {
		setAttrVal("Name_dep_out", Name_dep_out);
	}
	public String getName_dep_nur_out() {
		return ((String) getAttrVal("Name_dep_nur_out"));
	}	
	public void setName_dep_nur_out(String Name_dep_nur_out) {
		setAttrVal("Name_dep_nur_out", Name_dep_nur_out);
	}
	public String getName_bed_out() {
		return ((String) getAttrVal("Name_bed_out"));
	}	
	public void setName_bed_out(String Name_bed_out) {
		setAttrVal("Name_bed_out", Name_bed_out);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_orout";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_ap_out";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(OrdApOutDODesc.class);
	}
	
}